### PR TITLE
Support 3.4 and key/value list

### PR DESCRIPTION
**Pull request by hanachin**
チケット: https://insidemine.agileware.jp/issues/19085

### CODE REVIEW

File: app/controllers/importer_controller.rb [hunk 1]
```diff
@@ -731,13 +734,33 @@ def version_id_for_name!(project, name, add_versions)
...
   end
 
-  def process_multivalue_custom_field(project, add_versions, issue, custom_field, csv_val)
+  def enumeration_id_for_name!(custom_field, name, add_enumerations)
+    unless @enumeration_id_by_name.key?(name)
+      enumeration = custom_field.enumerations.find_by(name: name).try!(:id)
+      if enumeration.nil?
+        if name.try! { |n| n.size > 1 } && add_enumerations
+          max_position = custom_field.enumerations.pluck(:position).max || 0
+          enumeration = CustomFieldEnumeration.create(custom_field: custom_field, name: name, position: max_position + 1).id
```
Comment (hanachin) [inline app/controllers/importer_controller.rb]: TODO: remove position as well as 4.0

**Review by fkdkent — APPROVED** [review]

**Review by fkdkent — APPROVED** [review]

**Review by fkdkent — APPROVED** [review]

**Review by fkdkent — APPROVED** [review]

**Comment by fkdkent** [comment]
Oops! There is a failed test...

**Review by fkdkent — APPROVED** [review]

File: test/functional/importer_controller_test.rb [hunk 2]
```diff
@@ -20,64 +20,164 @@ def setup
...
+    post :result, build_params
+    assert_response :success
+    assert keyval_vals_for(Issue.find_by!(subject: 'パンケーキ')) == ['Tokyo']
+    assert keyval_vals_for(Issue.find_by!(subject: 'たこ焼き')) == ['Osaka']
+    issue = Issue.find_by!(subject: 'タピオカ')
+    assert ['Tokyo', 'Osaka'].all? { |area| area.in?(keyval_vals_for(Issue.find_by!(subject: 'タピオカ'))) }
+    assert Issue.find_by(subject: 'サーターアンダギー').nil?
+  end
+
+  test 'should error when assigned_to is missing' do
```
Comment (fkdkent) [inline test/functional/importer_controller_test.rb]: 🔗 https://github.com/agileware-jp/redmine_importer/pull/5/files#r316026246

File: test/functional/importer_controller_test.rb [hunk 3]
```diff
@@ -20,64 +20,164 @@ def setup
...
+    @issue.update!(assigned_to: @user)
+    post :result, build_params(update_issue: 'true').tap { |params| params[:fields_map]['assigned_to'] = 'assigned_to' }
+    assert_response :success
+    assert response.body.include?('Warning')
+    @issue.reload
+    assert_equal 'foobar', @issue.subject
+    assert_equal @user, @issue.assigned_to
+  end
+
+  test 'should error when assigned_to user is not assignable' do
```
Comment (fkdkent) [inline test/functional/importer_controller_test.rb]: 🔗 https://github.com/agileware-jp/redmine_importer/pull/5#discussion_r316026246

Comment (hanachin) [reply test/functional/importer_controller_test.rb]: That is correct, thank you 🙏

**Review by fkdkent — APPROVED** [review]

**Review by fkdkent — APPROVED** [review]

**Comment by kyamada23** [comment]
test ok

**Pull request closed without merging**